pkg/k8s/utils/utils_test.go

```diff
@@ -72,3 +75,171 @@ func TestServiceProxyName(t *testing.T) {
 		t.Fatalf("Expected test-svc-3, retrieved: %v", svcs)
 	}
 }
+
+func TestValidIPs(t *testing.T) {
+	tests := []struct {
+		name string
+		args slim_corev1.PodStatus
+		want []string
+	}{
+		{
+			name: "podip is nil",
+			args: slim_corev1.PodStatus{
+				HostIP: "127.0.0.1",
+			},
+			want: nil,
+		},
+
+		{
+			name: "one pod ip",
+			args: slim_corev1.PodStatus{
+				HostIP: "127.0.0.1",
+				PodIPs: []slim_corev1.PodIP{
+					{
+						IP: "127.0.0.2",
+					},
+				},
+			},
+			want: []string{"127.0.0.2"},
+		},
+
+		{
+			name: "duplicate ip",
+			args: slim_corev1.PodStatus{
+				HostIP: "127.0.0.1",
+				PodIPs: []slim_corev1.PodIP{
+					{
+						IP: "127.0.0.2",
+					},
+					{
+						IP: "127.0.0.2",
+					},
+				},
+			},
+			want: []string{"127.0.0.2"},
+		},
+		{
+			name: "multiple pod ip",
+			args: slim_corev1.PodStatus{
+				HostIP: "127.0.0.1",
+				PodIPs: []slim_corev1.PodIP{
+					{
+						IP: "10.0.0.1",
+					},
+					{
+						IP: "127.0.0.2",
+					},
+					{
+						IP: "127.0.0.3",
+					},
+				},
+			},
+			want: []string{"10.0.0.1", "127.0.0.2", "127.0.0.3"},
+		},
+		{
+			name: "have empty pod ip",
+			args: slim_corev1.PodStatus{
+				HostIP: "127.0.0.1",
+				PodIPs: []slim_corev1.PodIP{
+					{
+						IP: "127.0.0.2",
+					},
+					{
+						IP: "",
+					},
+				},
+			},
+			want: []string{"127.0.0.2"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := ValidIPs(tt.args); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("ValidIPs() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestIsPodRunning(t *testing.T) {
+	tests := []struct {
+		name string
+		args slim_corev1.PodStatus
+		want bool
+	}{
+		{
+			name: "Pod is not Running",
+			args: slim_corev1.PodStatus{
+				HostIP: "127.0.0.1",
+				Phase:  "Succeeded",
+			},
+			want: false,
+		},
+		{
+			name: "Pod is Running",
+			args: slim_corev1.PodStatus{
+				HostIP: "127.0.0.1",
+				Phase:  "Running",
+			},
+			want: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := IsPodRunning(tt.args); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("TestIsPodRunning() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestGetLatestPodReadiness(t *testing.T) {
```

Comment:
Can you generate some more realistic test data here? Pods often have many conditions, not just one.

Reply:
Updated, please review again.